checkov/yaml_doc/runner.py

```diff
@@ -53,3 +54,37 @@ def get_start_end_lines(
             start = result_config["__startline__"]
             end = result_config["__endline__"]
         return end, start
+
+
+def resolve_sub_name(definition: dict[str, Any], start_line: int, end_line: int, tag: str) -> str:
+    for key, sub_name in definition.get(tag, {}).items():
+        if key in [START_LINE, END_LINE]:
+            continue
+        if sub_name[START_LINE] <= start_line <= end_line <= sub_name[END_LINE]:
+            return str(key)
+    return ""
+
+
+def resolve_step_name(job_definition: dict[str, Any], start_line: int, end_line: int) -> str:
+    for idx, step in enumerate([step for step in job_definition.get('steps') or [] if step]):
+        if isinstance(step, str):
+            return f"{idx + 1}[{step}]"
+        elif isinstance(step, dict):
+            if step[START_LINE] <= start_line <= end_line <= step[END_LINE]:
+                name = step.get('name')
+                return f"{idx + 1}[{name}]" if name else str(idx + 1)
+    return ""
+
+
+def resolve_image_name(image_definition: dict[str, Any], start_line: int, end_line: int) -> str:
+    for idx, step in enumerate([step for step in image_definition.get('docker') or [] if step]):
+        if isinstance(image_definition.get('docker'), dict):
+            if step == 'image':
+                return f"{idx + 1}[{image_definition['docker'][step]}]"
+        if isinstance(step, str):
+            return f"{idx + 1}[{step}]"
+        elif isinstance(step, dict):
+            if step[START_LINE] <= start_line <= end_line <= step[END_LINE]:
+                name = step.get('image')
+                return f"{idx + 1}[{name}]" if name else str(idx + 1)
+    return ""
```

Comment:
IMO the image index isn't too useful to the resource name

The index is included for steps in GHA since they might not have a name, but we will always have the image name

Reply:
Since the image resource id is defined as `f'jobs.{job_name}.docker.image#{image_name}'`, as long as the same image appears on different jobs, the resource id will be distinct

Worth checking in circleci pipelines docs if using multiple images inside the same job is a valid use case

Reply:
an example from circleCI documentation: https://circleci.com/docs/configuration-reference/#docker
I will leave it numbered :)